arangod/RocksDBEngine/RocksDBIterators.cpp

```diff
@@ -333,3 +333,110 @@ void RocksDBSortedAllIterator::reset() {
   TRI_ASSERT(_trx->state()->isRunning());
   _iterator->Seek(_bounds.start());
 }
+
+
```

Comment:
I like the introduction of the generic iterator here. Do you think it would make any sense to adapt some of the other iterators to use this as a base as well? That's probably outside the scope of this PR, but could be added to our list of TODOs for refactoring/cleaning up the RocksDB engine.